packages/node/src/integrations/tracing/fastify/index.ts

```diff
@@ -143,15 +159,21 @@ function defaultShouldHandleError(_error: Error, _request: FastifyRequest, reply
  */
 export function setupFastifyErrorHandler(fastify: FastifyInstance, options?: Partial<FastifyHandlerOptions>): void {
   const shouldHandleError = options?.shouldHandleError || defaultShouldHandleError;
-
   const plugin = Object.assign(
     function (fastify: FastifyInstance, _options: unknown, done: () => void): void {
-      fastify.addHook('onError', async (request, reply, error) => {
-        if (shouldHandleError(error, request, reply)) {
-          captureException(error);
-        }
-      });
-
+      if (fastify.version?.startsWith('5.')) {
```

Comment:
This will break for > v5!

Reply:
This was intentional, just in case Fastify v6 uses a different diagnostics channel hook for errors when published. It won't break if `6.x` is used here, in the worst scenario, the error will be captured twice, and deduped eventually.

Reply:
> We can set a flag in dc hook when an error happens, but the error handler plugin will be registered already, then we can maybe unregister it

yeah lets give this a try